pkg/client/cache/delta_fifo.go

```diff
@@ -43,13 +45,13 @@ import (
 //       affects error retrying.
 //
 // Also see the comment on DeltaFIFO.
-func NewDeltaFIFO(keyFunc KeyFunc, compressor DeltaCompressor, knownObjectKeys KeyLister) *DeltaFIFO {
+func NewDeltaFIFO(keyFunc KeyFunc, compressor DeltaCompressor, knownObjects KeyLookup) *DeltaFIFO {
```

Comment:
I would really prefer that you not add the requirement to support GetByKey here.